lib/src/command/token_add.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// ignore_for_file: import_of_legacy_library_into_null_safe
+
+import 'dart:async';
+
+import '../authentication/token.dart';
+import '../command.dart';
+import '../io.dart';
+import '../log.dart' as log;
+import '../source/hosted.dart';
+
+/// Handles the `token add` pub command.
+class TokenAddCommand extends PubCommand {
+  @override
+  String get name => 'add';
+  @override
+  String get description => 'Add token for a server.';
```

Comment:
I prefer 'package repository'...

Reply:
> I prefer 'package repository'...

I prefer too, but this might cause conflict with packages hosted on git repository since 'repository' is a term in git ecosystem too..